### DISCUSSION TITLE
Is WM agnosticism truly feasible for LXQt?

### DISCUSSION THREAD

**Discussion by AzumaHazuki** [lxqt/lxqt#2144] [Q&A]
I appreciate and admire the commitment to agnosticism as far as possible regarding windowing systems. But, for a potential Wayland release especially, it seems less and less possible. 

Under Wayland, I see us having two options: KWin(FT) or LabWC, which is essentially Openbox reimplemented as a Wayland compositor. Under X11, we of course default to Openbox, though XFWM4 is also used and I know Tsujan uses Compiz-Reloaded and KWin as well with success.

That said, _most users are not power users._ This came up in previous bug reports/discussions, specifically about the WM selection dialog, which would scare off just about anyone. Given that LXQt has defaulted to Openbox from the start, that most distros pull it as a dependency anyway, and that we're _trying_ to remain lightweight, shouldn't we default to it? I don't necessarily mean making it a hard dependency, but something close maybe?

Personally I'd like to see KWin or KWinFT as the default WM, since we're already a Qt project and hard-depend on KWindowSystem. It's not ready yet (and there are a lot of problems with coherent theming, which I'll address in another topic...).

**Comment by tsujan** [comment]
> But, for a potential Wayland release especially, it seems less and less possible.

Yes, we know that. See recent comments at https://github.com/lxqt/lxqt/issues/10 for why we can't choose a Wayland WM/compositor yet. https://github.com/lxqt/lxqt/issues/10#issuecomment-821807530 summarizes the current situation. There are good comments there by people who work with Wayland (like the authors of KWinFT and Wayfire).

As for X11, LXQt will remain agnostic toward WMs/compositors. There are several good ones, like KWin, KWinFT, xfwm4, picom, compiz-reloaded,..., and even the old Openbox — all of them can be used with LXQt.
